src/USVPlugin.cpp

```diff
@@ -9,36 +8,34 @@ USVPlugin::~USVPlugin() {
     delete mWind;
     delete mThrusters;
     delete mActuators;
+    delete mDirectForce;
 }
 
-void USVPlugin::Load(physics::ModelPtr _model, sdf::ElementPtr _sdf)
+void USVPlugin::Load(physics::ModelPtr _model, sdf::ElementPtr _plugin_sdf)
 {
     mModel = _model;
-    // Import all thrusters from a model file (sdf)
-    sdf::ElementPtr modelSDF = mModel->GetSDF();
-    sdf::ElementPtr thrustersPluginElement = utilities::getPluginElementByName(
-        modelSDF, "thrusters"
-    );
-    // Since the wind plugin is optional and uses the same filename as the thrusters plugin, it should be found by name and be ignored if not found 
-    sdf::ElementPtr windPluginElement = utilities::findPluginElementByName(modelSDF, "wind_dynamics");
 
-    // Initialize communication node and subscribe to gazebo topic
     mNode = transport::NodePtr(new transport::Node());
     mNode->Init();
 
     mActuators = new Actuators(mModel, mNode);
-    mThrusters = new Thrusters;
 
     mWorldUpdateEvent = event::Events::ConnectWorldUpdateBegin(
         boost::bind(&USVPlugin::updateBegin, this, _1)
     );
 
-    loadThrusters(thrustersPluginElement);
-    loadRudders(thrustersPluginElement);
+    auto pluginName = _plugin_sdf->Get<std::string>("name");
+    if ("thrusters" == pluginName) {
+        mThrusters = loadThrusters(_plugin_sdf);
+        mRudders = loadRudders(_plugin_sdf);
+    }
 
```

Comment:
Do not add spaces between if and else

Generally speaking, don't invent or adopt style that you are accustomed to when it is obviously not the style that is in use in the current code base. Being inconsistent is a lot worse than being "right".

Reply:
no problem, done